### PR TITLE
refactor: add partition num as param in partition lock's init_fn

### PR DESCRIPTION
**Pull request by tanruixiang**
## Detailed Changes
- Add a param(partition_num) in partition lock's init_fn.

## Test Plan
UT

### CODE REVIEW

File: components/object_store/src/disk_cache.rs [hunk 1]
```diff
@@ -292,14 +297,10 @@ impl DiskCacheStore {
         underlying_store: Arc<dyn ObjectStore>,
         partition_bits: usize,
     ) -> Result<Self> {
-        ensure!(
-            cap % (page_size * (1 << partition_bits)) == 0,
-            InvalidCapacity
-        );
-        let cap_per_part = cap / page_size / (1 << partition_bits);
-        ensure!(cap_per_part != 0, InvalidCapacity);
+        let pagenum_per_part = cap / page_size;
```
Comment (jiacai2050) [inline components/object_store/src/disk_cache.rs:300]: page_num_per_part

File: components/object_store/src/disk_cache.rs [hunk 2]
```diff
@@ -124,13 +124,18 @@ struct DiskCache {
...
+            ensure!(cap_per_par != 0, InvalidCapacity);
+            Ok(LruCache::new(cap / partition_num))
+        };
+
+        Ok(Self {
             root_dir,
-            cap,
-            cache: PartitionedMutexAsync::new(init_lru, partition_bits, SeaHasherBuilder {}),
-        }
+            cap: cap / (1 << partition_bits),
```
Comment (jiacai2050) [inline components/object_store/src/disk_cache.rs:136]: This is different with before, isn't just `cap` enough?

Comment (tanruixiang) [reply components/object_store/src/disk_cache.rs:136]: This value is the capacity that each partition's LRU can hold and will be used for LRU.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**